### ISSUE TITLE
MockServer not working

### ISSUE THREAD

**Issue by mtitolo** [thoov/mock-socket#174]
**Tl;DR** the Mock Server isn't working. I've reproduced in a sample project: https://github.com/mtitolo/test-mock-websockets

I've tried this on node 7 and node 8. I'm using version 7.0.0 of the library on macOS 10.12.6. I've tried using different names for "localhost" (127.0.0.1 is the one in the sample), and nada. The mock socket doesn't open.

I've tried debugging myself. The issue does not seem to be with the actual standing up of the mock server. All of those functions fire appropriately. Rather I think the issue is with the intercepting. The function that doesn't fire is `EventTarget.dispatchEvent` on the Server. Everything else seems to be working.

**Comment by mtitolo** [comment]
More digging!

```
 /*
  * Attaches the mock websocket object to the global object
  */
  Server.prototype.start = function start () {
    var globalObj = retrieveGlobalObject();
    console.log("Starting mock server")

    if (globalObj.WebSocket) {
      console.log("found a global websocket object")
      this.originalWebSocket = globalObj.WebSocket;
    }

    globalObj.WebSocket = WebSocket$1;
  };
```

The console will have `Starting mock server` but not `found a global websocket object`. Now to figure out _why_.

**Comment by thoov** [comment]
Hi @mtitolo. Sorry for the delay. I have been without internet for the last week and the struggle has been real :-). Also sorry you are running into this issue and thank you for creating that sample repo. I believe I know what is going on. The short story is that the ws websocket is not being correctly mocked. This is currently something that this project doesn't have a great story around but I am thinking of a better dev solution for this. I will submit a PR to your sample repo that will show you how we can work around this issue. I should have that PR done either later today or tonight.
